BossMod/Components/BaitAway.cs

```diff
@@ -178,22 +178,45 @@ public override void OnUntethered(BossModule module, Actor source, ActorTetherIn
     public class BaitAwayCast : GenericBaitAway
     {
         public AOEShape Shape;
+        public bool ResolvesAtCastEvent;
```

Comment:
Not a fan of this one in the generic components, it's too easy to misuse by mistake, without understanding all implications. There are good reasons why cast-start should generally be paired with cast-finish - each cast-start is _guaranteed_ to have a corresponding cast-finish (eg if spell is interrupted or canceled, if actor is destroyed, etc - and actor destruction can happen for a number of reasons), and obviously there is no such guarantee for cast-event.

Are there any examples other than sugriva that need this? If not, I'd prefer if it stayed specific to that component, with the comment describing why it is done the way it is done, and only promoted to generic component once we have evidence that this is a mechanic of the game (and then maybe something similar would be needed for other casts...)

Reply:
well there are still 100s of encounters without a module and I doubt that they made this mechanic just for one open world boss. but i understand your point.